### PR TITLE
Migrating to .NET 8

### PR DESCRIPTION
**Pull request by stsrki**

### CODE REVIEW

**Comment by stsrki** [comment]
@David-Moreira it seems that the Docs project is now working in -NET 8 RC2. I cannot find any issue so far. Please review once you have tome.

File: Documentation/Blazorise.Docs.Server/Startup.cs [hunk 1]
```diff
@@ -33,14 +34,29 @@ public Startup( IConfiguration configuration )
...
+        // Add services to the container.
+        services
+            .AddRazorComponents()
+            .AddInteractiveServerComponents();
 
-        services.AddServerSideBlazor().AddHubOptions( ( o ) =>
+        services.Configure<IISServerOptions>( options =>
         {
-            o.MaximumReceiveMessageSize = 1024 * 1024 * 100;
+            options.AllowSynchronousIO = true;
```
Comment (David-Moreira) [inline Documentation/Blazorise.Docs.Server/Startup.cs:44]: Will you remember why this was explicitly set? If not better add a comment.

File: Documentation/Blazorise.Docs.Server/Startup.cs [hunk 2]
```diff
@@ -33,14 +34,29 @@ public Startup( IConfiguration configuration )
     // For more information on how to configure your application, visit https://go.microsoft.com/fwlink/?LinkID=398940
     public void ConfigureServices( IServiceCollection services )
     {
-        services.AddRazorPages();
-        services.AddServerSideBlazor();
+        // Add services to the container.
+        services
+            .AddRazorComponents()
+            .AddInteractiveServerComponents();
```
Comment (David-Moreira) [inline Documentation/Blazorise.Docs.Server/Startup.cs:41]: Want to try auto mode? By default docs opens in Server mode, but when dlls are available it opens in wasm.

File: Documentation/Blazorise.Docs.Server/Startup.cs [hunk 3]
```diff
@@ -33,14 +34,29 @@ public Startup( IConfiguration configuration )
...
+        } );
+
+        //services.AddRazorPages();
+        //services.AddServerSideBlazor();
+
+        //services.AddServerSideBlazor().AddHubOptions( ( o ) =>
+        //{
+        //    o.MaximumReceiveMessageSize = 1024 * 1024 * 100;
+        //} );
+
```
Comment (David-Moreira) [inline Documentation/Blazorise.Docs.Server/Startup.cs:59]: just remove visual thrash

File: Documentation/Blazorise.Docs/App.razor [hunk 4]
```diff
@@ -1,78 +1,65 @@
...
+
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="utf-8" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+
+    <base href="/" />
+
+    <HeadOutlet />
```
Comment (David-Moreira) [inline Documentation/Blazorise.Docs/App.razor:13]: Does this need to be interactive?

File: Documentation/Blazorise.Docs/Pages/Home/Counter.razor [hunk 5]
```diff
@@ -0,0 +1,18 @@
+﻿@page "/counter"
```
Comment (David-Moreira) [inline Documentation/Blazorise.Docs/Pages/Home/Counter.razor:1]: remove page

File: Documentation/Blazorise.Docs/wwwroot/blazorise.docs.js [hunk 6]
```diff
@@ -17,6 +17,10 @@
         },
 
         generateToc: (targetElement, options) => {
+            if (!document.getElementById('TableOfContents')) {
```
Comment (David-Moreira) [inline Documentation/Blazorise.Docs/wwwroot/blazorise.docs.js:20]: Why do we need this?

File: NuGet.config [hunk 7]
```diff
@@ -0,0 +1,5 @@
+<configuration>
+  <packageSources>
+    <add key="dotnet8" value="https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet8/nuget/v3/index.json" />
```
Comment (David-Moreira) [inline NuGet.config:3]: do we need this?

**Review by David-Moreira — APPROVED** [review]
Code LGTM. I haven't tested the application, It's late here.
But if it works just like before, then great!

File: Documentation/Blazorise.Docs/wwwroot/blazorise.docs.js (hunk 6 above)
Comment (stsrki) [reply Documentation/Blazorise.Docs/wwwroot/blazorise.docs.js:20]: This is the navigation on the right side

image

**Pull request merged**